dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NumberBoxParser.h"
+#include "Utils.h"
+
+static constexpr wstring_view c_numberBoxOperators{ L"+-*/^"sv };
+
+// Returns list of MathTokens from expression input string. If there are any parsing errors, it returns an empty vector.
+std::vector<MathToken> NumberBoxParser::GetTokens(const wchar_t* input, winrt::INumberParser numberParser)
+{
+    auto tokens = std::vector<MathToken>();
+
+    bool expectNumber = true;
+    while (input[0] != '\0')
+    {
+        // Skip spaces
+        auto nextChar = input[0];
+        if (nextChar != L' ')
+        {
+            if (expectNumber)
+            {
+                if (nextChar == '(')
+                {
+                    // Open parens are also acceptable, but don't change the next expected token type.
+                    tokens.push_back(MathToken(MathTokenType::Parenthesis, nextChar));
+                }
+                else
+                {
+                    const auto [value, charLength] = GetNextNumber(input, numberParser);
+
+                    if (charLength > 0)
+                    {
+                        tokens.push_back(MathToken(MathTokenType::Numeric, value));
+                        input += charLength - 1; // advance the end of the token
+                        expectNumber = false; // next token should be an operator
+                    }
+                    else
+                    {
+                        return {};
```

Comment:
>return {}; [](start = 24, length = 10)

I would keep the comments describing the error.

Reply:
Fixed.